app/mutations/addresses/release.rb

```diff
@@ -31,10 +32,18 @@ def execute
         info "Skip gateway address=#{address.id} in pool=#{@pool.id}"
 
       else
-        info "Delete address=#{address.id} in pool=#{@pool.id}"
-
-        address.delete!
+        if ping?(address.address)
+          info "#{self.address} still responds to ping, deleting omitted"
```

Comment:
would 202-Accepted work?
> 202 Accepted
> The request has been accepted for processing, but the processing has not been completed. The request might or might not be eventually acted upon, and may be disallowed when processing occurs.

IMO 4xx are "true" errors and this case is now something expected and not a real error :)